### ISSUE TITLE
Documentation for JavaScript implementation?

### ISSUE THREAD

**Issue by benkahandevelopment** [bradmartin/nativescript-lottie#6]
I'm really excited to use this plugin but I can't yet get it to work in a JavaScript (not TS) environment.

I have this loader object defined in XML:

    <Page
        navigatedTo="navigatedTo"
        xmlns="http://schemas.nativescript.org/tns.xsd"
        xmlns:Lottie="nativescript-lottie">

        <StackLayout height="300" class="loader-container" horizontalAlignment="center" verticalAlignment="center">
            <Lottie:LottieView height="150" id="lottieLoader" src="lottieLoader.json" loop="true" autoplay="true" loaded="lottieLoaded" />
        </StackLayout>
    </Page>

And the `lottieLoader.json` in the `/platforms/android/main/assets/` folder. I always rebuild and uninstall/reinstall on device/emulator before test, but the object never appears on the page (or even takes up any page space).

I try to catalogue the events, but I can't link the loader object in XML to the JavaScript:

    var lottieModule = require("nativescript-lottie").LottieView;
    var lottieLoader;

    exports.lottieLoaded = function(args){
        lottieLoader = new lottieModule(args.object);
        console.log(lottieLoader.isAnimating());
    }

In this above example, the emulator (not the console) outputs on the screen:

> Cannot read property 'isAnimating' of undefined, undefined

Can we please have a JavaScript example? Where am I going wrong in my code?

**Comment by mudlabs** [comment]
Not really addressing the issue of creating a JS example here. But I have got it going in Android, without using any JavaScript though.

**For Android:**

1. I created an assets directory in app to store the lottie files. (i.e. `app/assets/lottieAnimation.json`)
2. In the `xml` file I use `android:src="app/assets/lottieAnimation.json"`
3. And that's it, Lottie file just renders and does its thing. No Javascript/code-behind needed.

**For iOS**
I haven't got iOS operating yet though. The `LottieView` renders, but it's just empty, white space.

One warning I get in terminal after `tns run ios` is that there's no platform specified, so `ios version 8.0` was auto assigned. And that I should `specify a platform for this target in your Podfile`. But I can't even find said Podfile.

If it means to add one to `platforms/ios/` how do I do that? It's my understanding the platforms directory is like the `_site` directory in Jekyll. You don't touch it because it's just going to be re-build all the time.

**Comment by mudlabs** [comment]
I'm having issues getting it to render on `iOS`. But see how the following works for you as a JS implementation.

**Say you had this xml**
```xml
<Page xmlns="http://schemas.nativescript.org/tns.xsd" navigatedTo="pageNavigatedTo">
  <Page.actionBar>
    <ActionBar>
      <Label text="JS only Lottie"></Label>
    </ActionBar>
  </Page.actionBar>
  <StackLayout id="stackView"></StackLayout>
</Page>
```

**Then your code behind**
```js
const lottieModule = require("nativescript-lottie");

exports.pageNavigatedTo = args => {
  const page = args.object;
  const stackView = page.getViewById("stackView");
  const lottie = new lottieModule.LottieView();
  
  lottie.set("src", "res://assets/lottieThing.json");
  lottie.set("height", 130);
  lottie.set("loop", true);
  lottie.set("autoPlay", true);
  // I'm only setting backgroundColor here because of the rendering issue I'm having.
  lottie.set("backgroundColor", "#777");
  
  stackVeiw.addChild(lottie);
}
```

How does that work for you @benkahandevelopment ?

I can confirm the `LottieView` renders to the `xml` file as I see the grey rectangle appear. And when inspecting via the Chrome DevTools I can see it included in the tree. But I'm unable to confirm if the Lottie file actually loads and animates.

**Comment by bradmartin** [comment]
That sample looks great, I think the path to the `.json` needs tweaking. A month or two ago @rhanb worked on a hook that would copy assets into the android `assets` dir I believe. We should probably add that as a dependency of this plugin and specify a path for lottie .json files to reside in users app to make sure the hook picks them up.

**Comment by rhanb** [comment]
Yep this [hook](https://github.com/rhanb/nativescript-dev-assets) allows you to place your `.json` files into your app folder. When you install it, it creates an `assets` folder, all your assets should be in there. Then you just have to use the filename as an `src` 😄

**Comment by benkahandevelopment** [comment]
@bradmartin @rhanb @mudlabs thanks for all your input. I've just tried the following with no luck:

First, I installed the hook `nativescript-dev-assets`, and placed `deer.json` in the `app\assets\` folder.

Here is my `main-page.xml`:

```
<Page
    xmlns="http://schemas.nativescript.org/tns.xsd"
    xmlns:Lottie="nativescript-lottie"
    navigatedTo="onNavigatedTo">
    <StackLayout id="lottieCont"></StackLayout>
</Page>
```

And my `main-page.js`:

```
var createViewModel = require("./main-view-model").createViewModel;
var lottieModule = require("nativescript-lottie");
exports.onNavigatedTo = args => {
    const page = args.object;
    const lottieCont = page.getViewById("lottieCont");
    const lottie = new lottieModule.LottieView();
    lottie.set("src", "deer.json");
    lottie.set("height", 130);
    lottie.set("loop", true);
    lottie.set("backgroundColor", "#777");
    lottieCont.addChild(lottie);
}
```

However, this is what the emulator shows (after building/running successfully). As you can see, I can't even render the grey background on the `StackView` element:

![image](https://user-images.githubusercontent.com/1959420/34906061-0984133e-f85d-11e7-9b7f-1ffb5c56a634.png)

I have also tried changing this line with no luck:

```
//From
lottie.set("src", "deer.json");
//To
lottie.set("src", "res://assets/deer.json");
lottie.set("src", "res://deer.json");
lottie.set("src", "assets/deer.json");
```

Can you see what I'm doing wrong here?

**Comment by rhanb** [comment]
@benkahandevelopment if you use {N} version > `4.1.x` please make sure to have the latest version of `nativescript-dev-assets` (0.5).

You don't have any error output in your console? Thanks

**Comment by rhanb** [comment]
@bradmartin I added `nativescript-dev-assets` as a dependency like this:

```json
"scripts": {
    "postinstall": "cd ../../ && tns install nativescript-dev-assets"
}
```
If a simply add `nativescript-dev-assets` to the `devDependencies`, the `postinstall` script of `nativescript-dev-assets` will be executed in the wrong context (inside `nativescript-lottie` folder).

**Comment by mudlabs** [comment]
Sorry @benkahandevelopment I could not reproduce your issue.

Have you tried using `file-system` to check that `deer.json` exists at the path you expect?
```javascript
const fileSystem = require("file-system");

exports.onNavigatedTo = args => {
  const path = "app/assets/deer.json";
  fileSystem.knownFolders.currentApp().contains(path)
    ? console.log(`Yes deer.json is at path: ${path}`)
    : console.error(`No deer.json is not at path: ${path}`);
};
```
**Note** If your adding the Lotties via code-behind you don't need the `xmlns:Lottie="nativescript-lottie"` attribute attached to `Page` . Though that shouldn't been the problem here.

**Comment by benkahandevelopment** [comment]
@rhanb I have added the plugin `nativescript-dev-assets` and there is still no result. I saw your other response - does this need to be added into the `package.json` file too or something? Or is adding the plugin from the command line with `tns plugin add nativescript-dev-assets`  enough?

@mudlabs the code shows the following:

*Input*
```
const path = "app/assets/deer.json";
fileSystem.knownFolders.currentApp().contains(path) ? console.log(`Yes, deer.json is at path ${path}`) : console.error(`No, deer.json is not at path ${path}`);
```

*Output*
```
JS: No, deer.json is not at path app/assets/deer.json
````

Here is what my directory structure looks like, and the location of `deer.json`:

![image](https://user-images.githubusercontent.com/1959420/36343838-07751422-1409-11e8-9c43-5d8fbc4ca3e9.png)

What are your thoughts as to why this isn't working?
